src/activity_handlers.cpp

```diff
@@ -295,27 +298,28 @@ void set_up_butchery( player_activity &act, player &u, butcher_type action )
     }
     // workshop butchery (full) prequisites
     if( action == BUTCHER_FULL ) {
-            if( !g->m.has_flag_furn( "BUTCHER_EQ", u.pos() ) ) {
-                u.add_msg_if_player( m_info, _( "You need a butchering rack to perform a full butchery." ) );
+            if( corpse_item.get_mtype()->size >= MS_MEDIUM && !g->m.has_flag_furn( "BUTCHER_EQ", u.pos() ) ) {
```

Comment:
Does `!g->m.has_flag_furn( "BUTCHER_EQ", u.pos() )` mean this still requires you to be standing on top of a furniture with the "BUTCHER_EQ" flag?

Reply:
`corpse_item.get_mtype()->size` can be changed to `corpse->size`, or `corpse.size` if changed to a const reference.

Reply:
Ad 1. yes, as it checks `u.pos()` where `u` is You.